src/GdbConnection.cc

```diff
@@ -1654,7 +1654,7 @@ void GdbConnection::reply_get_thread_list(const vector<GdbThreadId>& threads) {
         1 /*m char*/ +
         threads.size() * (1 /*p*/ + 2 * sizeof(threads[0]) + 1 /*,*/) +
         1 /*\0*/;
-    char* str = new char*[maxlen];
+    char* str = new char(maxlen);
```

Comment:
This is still wrong, isn't it? This is initializing a `char` with `maxlen`, but not allocating `maxlen` chars.